src/multilayerqg.jl

```diff
@@ -455,42 +504,116 @@ Compute the inverse Fourier transform of `varh` and store it in `var`.
 invtransform!(var, varh, params::AbstractParams) = ldiv!(var, params.rfftplan, varh)
 
 """
-    streamfunctionfrompv!(ψh, qh, params, grid)
+    pvfromstreamfunction!(qh, ψh, params, grid)
 
-Inverts the PV to obtain the Fourier transform of the streamfunction `ψh` in each layer from
-`qh` using `ψh = params.S⁻¹ qh`.
+Obtain the Fourier transform of the PV from the streamfunction `ψh` in each layer using 
+`qh = params.S * ψh`.
 """
-function streamfunctionfrompv!(ψh, qh, params, grid)
+function pvfromstreamfunction!(qh, ψh, params, grid)
   for j=1:grid.nl, i=1:grid.nkr
-    CUDA.@allowscalar @views ψh[i, j, :] .= params.S⁻¹[i, j] * qh[i, j, :]
+    CUDA.@allowscalar @views qh[i, j, :] .= params.S[i, j] * ψh[i, j, :]    
   end
   
   return nothing
 end
 
 """
-    pvfromstreamfunction!(qh, ψh, params, grid)
+    pvfromstreamfunction!(qh, ψh, params::SingleLayerParams, grid)
 
-Obtains the Fourier transform of the PV from the streamfunction `ψh` in each layer using 
-`qh = params.S * ψh`.
+Obtain the Fourier transform of the PV from the streamfunction `ψh` for the special
+case of a single fluid layer configuration. In this case, ``q̂ = - k² ψ̂``.
 """
-function pvfromstreamfunction!(qh, ψh, params, grid)
+function pvfromstreamfunction!(qh, ψh, params::SingleLayerParams, grid)
+  @. qh = -grid.Krsq * ψh
+  
+  return nothing
+end
+
+"""
+    pvfromstreamfunction!(qh, ψh, params::TwoLayerParams, grid)
+
+Obtain the Fourier transform of the PV from the streamfunction `ψh` for the special
+case of a two fluid layer configuration. In this case we have,
+
+```math
+q̂₁ = - k² ψ̂₁ + f₀² / (g′ H₁) * (ψ̂₂ - ψ̂₁) ,
+```
+
+```math
+q̂₂ = - k² ψ̂₂ + f₀² / (g′ H₂) * (ψ̂₁ - ψ̂₂) .
+```
+
+(Here, the PV-streamfunction relationship is hard-coded to avoid scalar operations
+on the GPU.)
+"""
+function pvfromstreamfunction!(qh, ψh, params::TwoLayerParams, grid)
+  f₀, g′, H₁, H₂ = params.f₀, params.g′, params.H[1], params.H[2]
+  
+  ψ1h, ψ2h = ψh[:, :, 1], ψh[:, :, 2]
+
+  @. qh[:, :, 1] = - grid.Krsq * ψ1h + f₀^2 / (g′ * H₁) * (ψ2h - ψ1h)
+  @. qh[:, :, 2] = - grid.Krsq * ψ2h + f₀^2 / (g′ * H₂) * (ψ1h - ψ2h)
```

Comment:
Is there no need for `@views` here?

Reply:
Not sure. Why should we?

Reply:
Adding @views in both `streamfunctionfrompv!` and `pvfromstreamfunction!` reduced the allocations but only by just a bit. There are still some allocations (see, e.g., the comment at the top of the PR). Should we worry?

Reply:
e.g., 
```julia
julia> nlayers = 2; prob = MultiLayerQG.Problem(nlayers, GPU()); @btime stepforward!(prob)
2.026 ms (3713 allocations: 326.77 KiB)
```
after `@views` became
```julia
julia> nlayers = 2; prob = MultiLayerQG.Problem(nlayers, GPU()); @btime stepforward!(prob)
1.923 ms (3385 allocations: 321.58 KiB)
```